CMakeLists.txt

```diff
@@ -26,18 +26,14 @@ if(CMAKE_SIZEOF_VOID_P MATCHES "8" AND NOT(FORCE_32))
     set(ARCH_64 1)
 endif()
 
-if(CMAKE_BUILD_TYPE STREQUAL "Debug")
-    add_definitions(-D_DEBUG)
-    if (NOT WIN32)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g")
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g")
-    endif()
-else()
-    if (NOT WIN32)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2")
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2")
-    endif()
-    add_definitions(-DNDEBUG)
+if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_CXX_COMPILER_ID MATCHES "^(Apple)?Clang$")
+    # override default -O3
+    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O2")
+    set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O2")
```

Comment:
To make CMAKE_C_FLAGS_RELEASE and CMAKE_CXX_FLAGS_RELEASE work could you please add the RELEASE type in buildall.py:
```
-            ("-DCMAKE_BUILD_TYPE=Debug" if args.debug else ""),
+            ("-DCMAKE_BUILD_TYPE=Debug" if args.debug else "-DCMAKE_BUILD_TYPE=RELEASE"),
```

Reply:
Does it really overrides the -O3 flag?
    # override default -O3
    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O2")
    set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O2")
    
I just tried to build release libs and it looks like we have both flags (-O3 and -O2) now:
  
 ```
[ 60%] Building C object CMakeFiles/ittnotify.dir/src/ittnotify/ittnotify_static.c.o
/usr/bin/cc  -I/localdisk/ittapi_git/ittapi/include -I/localdisk/ittapi_git/ittapi/src/ittnotify  -O3 -DNDEBUG -O2 -fPIC   -o CMakeFiles/ittnotify.dir/src/ittnotify/ittnotify_static.c.o   -c /localdisk/ittapi_git/ittapi/src/ittnotify/ittnotify_static.c
[ 80%] Building C object CMakeFiles/ittnotify.dir/src/ittnotify/jitprofiling.c.o
/usr/bin/cc  -I/localdisk/ittapi_git/ittapi/include -I/localdisk/ittapi_git/ittapi/src/ittnotify  -O3 -DNDEBUG -O2 -fPIC   -o CMakeFiles/ittnotify.dir/src/ittnotify/jitprofiling.c.o   -c /localdisk/ittapi_git/ittapi/src/ittnotify/jitprofiling.c
```

Reply:
@ilya-lavrenov thank you much for the detailed description for the change!
now we just need to make RELEASE flags work and override the -O3 flag with -O2

Reply:
Fixed by adding `Release` as a default configuration.